lib/std/crypto/tlcsprng.zig

```diff
@@ -72,6 +72,11 @@ fn tlsCsprngFill(_: *const std.rand.Random, buffer: []u8) void {
         return fillWithOsEntropy(buffer);
     }
 
+    // Use OS syscall for haiku
+    if (std.Target.current.os.tag == .haiku) {
+        return fillWithOsEntropy(buffer);
+    }
```

Comment:
Here are some questions to help decide the best implementation here:
 * does haiku support fork() ?
   * if so, does haiku support MADV_WIPEONFORK?
 * does haiku support `arc4random_buf` or equivalent? or is /dev/urandom the only way to get entropy from the OS?
 * does haiku support threadlocal variables?

Reply:
Appreciate the feedback on this one. I have some doubts myself on this current implementation as-is but do know that it is functional.

From what I can tell, Haiku does support fork() although not intricately familiar with its implementation of fork. I don't think it supports MADV_WAVEONFORK from my short look for it. I can try to dig in a little more and see if it does support it.

Haiku does have appear to have an arc4random() call available as part of its FreeBSD network compatibility layer (think an extra lib will need to be linked) but am unsure if this is something that is a general usage or more tailored to work for Haiku's network stack. My guess is that /dev/urandom is not the only mechanism available (https://dev.haiku-os.org/ticket/14937 might be useful)
 
As for threadlocal variables, I am not sure what support has for this. I am going to spend some more time to look into this.

Thanks for the feedback! I am going to spend some more time to think about if there are better approaches to handling this.

Reply:
If it helps - the best case scenario is that haiku has arc4random_buf or equivalent. If that is the case, there is an easy solution to this implementation.